daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -1191,23 +1229,85 @@ def forward(node: ONNXOp, state: SDFGState,
         #empty memlet
         new_state.add_memlet_path(write_max_res, vect_mx, memlet=dace.Memlet())
         #Attention, the storing location must take into account that the input was vectorized
-        if vec_width != 1:
+        if vec_width != 1 and Y.veclen == 1:
```

Comment:
Given the checks in `can_be_applied` this should not occur

Reply:
It can occur, `vec_width` is the input vectorization i.e. `X.veclen`.

Assume `Y.veclen == 1` then the testing block:
 ```
if Y.veclen != 1:  # if output vectorized must match
            _, filter_width = node.kernel_shape
            if not (X.veclen == Y.veclen or X.veclen == filter_width * Y.veclen):  # support reducing vector size proportionally to filter
                return False
```
Will be skipped and thus all cases for `X.veclen` are allowed i.e. `vec_width == 1` and `vec_width != 1`

If `Y.veclen != 1` then the testing block will be executed and enforces a *input* vectorization width (`X.veclen`) which is either the same or a `filter_width` multiple of the output.

Reply:
I could refactor `vec_width` to `in_vec_width` to make this more explicit, if you like?

Reply:
No probl, I missed the conditions in the `can_be_applied`